Error while checking build/O2Physics/o2 for 3283b9a00314b9609a168a5757349af3e4be0401 at 2026-08-10 12:50:
```
## sw/BUILD/O2Physics-latest/log
c++: fatal error: Killed signal terminated program cc1plus
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/17388/3283b9a00314b9609a168a5757349af3e4be0401/build_O2Physics_o2/pretty.html).

--- a/PWGCF/Femto/Core/collisionBuilder.h
+++ b/PWGCF/Femto/Core/collisionBuilder.h
@@ -245,7 +245,13 @@ class CollisionSelection : public baseselection::BaseSelection<float, o2::analys
 
     // event shape
     mQvecDetector = static_cast<modes::QvecDetector>(config.qvecDetector.value);
+    if (mQvecDetector >= modes::QvecDetector::kQvecDetectorLast) {
+      LOG(fatal) << "Qvector Detector is not supported";
+    }
     mQvecHarmonic = static_cast<modes::QvecHarmonic>(config.qvecHarmonic.value);
+    if (mQvecHarmonic < modes::QvecHarmonic::kN2 || mQvecHarmonic >= modes::QvecHarmonic::kQvecHarmonicLast) {
+      LOG(fatal) << "Qvector Harmonic is not supported";
+    }
 
     this->addSelection(kSel8, collisionSelectionNames.at(kSel8), config.sel8.value);
     this->addSelection(kNoSameBunchPileUp, collisionSelectionNames.at(kNoSameBunchPileUp), config.noSameBunchPileup.value);
@@ -371,9 +377,6 @@ class CollisionSelection : public baseselection::BaseSelection<float, o2::analys
   {
     auto harmonic = static_cast<float>(mQvecHarmonic);
     int index = static_cast<int>(mQvecHarmonic) - 2; // get index in the qvector vector
-    if (index >= 2) {
-      LOG(fatal) << "At the moment harmonics up to 3 are supported!";
-    }
     switch (mQvecDetector) {
       case modes::QvecDetector::kFT0C:
         mEventPlane = RecoDecay::constrainAngle((std::atan2(col.qvecFT0CImVec()[index], col.qvecFT0CReVec()[index])) / harmonic, 0, harmonic); // constrain between 0 and 2pi/harmonic

--- a/PWGCF/Femto/Core/collisionBuilder.h
+++ b/PWGCF/Femto/Core/collisionBuilder.h
@@ -82,7 +82,7 @@ struct ConfCollisionBits : o2::framework::ConfigurableGroup {
   o2::framework::Configurable<std::vector<float>> sphericityMax{"sphericityMax", {}, "Maximum sphericity"};
   o2::framework::Configurable<std::vector<std::string>> triggers{"triggers", {}, "List of all triggers to be used"};
   o2::framework::Configurable<datatypes::QvecDetectorType> qvecDetector{"qvecDetector", 0, "Detector used to estimate the Q-vector: 0 -> FT0C, 1 -> FT0A"};
-  o2::framework::Configurable<datatypes::QvecHarmonicType> qvecHarmonic{"qvecHarmonic", 2, "Harmonic n of the Q-vector and event plane angle Psi_n: 1 -> direct, 2 -> elliptic, 3 -> triangular"};
+  o2::framework::Configurable<datatypes::QvecHarmonicType> qvecHarmonic{"qvecHarmonic", 2, "Harmonic n of the Q-vector and event plane angle Psi_n: 2 -> elliptic, 3 -> triangular"};
 };
 
 struct ConfCcdb : o2::framework::ConfigurableGroup {
@@ -370,12 +370,16 @@ class CollisionSelection : public baseselection::BaseSelection<float, o2::analys
   void setEventPlane(T const& col)
   {
     auto harmonic = static_cast<float>(mQvecHarmonic);
+    int index = static_cast<int>(mQvecHarmonic) - 2; // get index in the qvector vector
+    if (index >= 2) {
+      LOG(fatal) << "At the moment harmonics up to 3 are supported!";
+    }
     switch (mQvecDetector) {
       case modes::QvecDetector::kFT0C:
-        mEventPlane = RecoDecay::constrainAngle((std::atan2(col.qvecFT0CImVec()[0], col.qvecFT0CReVec()[0])) / harmonic, 0, harmonic); // constrain between 0 and 2pi/harmonic
+        mEventPlane = RecoDecay::constrainAngle((std::atan2(col.qvecFT0CImVec()[index], col.qvecFT0CReVec()[index])) / harmonic, 0, harmonic); // constrain between 0 and 2pi/harmonic
         break;
       case modes::QvecDetector::kFT0A:
-        mEventPlane = RecoDecay::constrainAngle((std::atan2(col.qvecFT0AImVec()[0], col.qvecFT0AReVec()[0])) / harmonic, 0, harmonic); // constrain between 0 and 2pi/harmonic
+        mEventPlane = RecoDecay::constrainAngle((std::atan2(col.qvecFT0AImVec()[index], col.qvecFT0AReVec()[index])) / harmonic, 0, harmonic); // constrain between 0 and 2pi/harmonic
         break;
     }
   }

--- a/PWGCF/Femto/Core/collisionBuilder.h
+++ b/PWGCF/Femto/Core/collisionBuilder.h
@@ -368,6 +368,12 @@ class CollisionSelection : public baseselection::BaseSelection<float, o2::analys
       case modes::QvecDetector::kFT0A:
         mQvec = std::hypot(col.qvecFT0AReVec()[0], col.qvecFT0AImVec()[0]) * std::sqrt(col.sumAmplFT0A());
         break;
+      case modes::QvecDetector::kQvecDetectorLast:
+        LOG(fatal) << "Invalid Q-vector detector";
+        break;
+      default:
+        LOG(fatal) << "Invalid Q-vector detector";
+        break;
     }
   }
   [[nodiscard]] float getQvector() const { return mQvec; }
@@ -383,6 +389,12 @@ class CollisionSelection : public baseselection::BaseSelection<float, o2::analys
         break;
       case modes::QvecDetector::kFT0A:
         mEventPlane = RecoDecay::constrainAngle((std::atan2(col.qvecFT0AImVec()[index], col.qvecFT0AReVec()[index])) / harmonic, 0, harmonic); // constrain between 0 and 2pi/harmonic
+        break;
+      case modes::QvecDetector::kQvecDetectorLast:
+        LOG(fatal) << "Invalid Q-vector detector";
+        break;
+      default:
+        LOG(fatal) << "Invalid Q-vector detector";
         break;
     }
   }

--- a/PWGCF/Femto/Core/modes.h
+++ b/PWGCF/Femto/Core/modes.h
@@ -184,7 +184,6 @@ enum class QvecDetector : o2::analysis::femto::datatypes::QvecDetectorType {
 };
 
 enum class QvecHarmonic : o2::analysis::femto::datatypes::QvecHarmonicType {
-  kN1 = 1,
   kN2 = 2,
   kN3 = 3
 };

--- a/PWGCF/Femto/Core/modes.h
+++ b/PWGCF/Femto/Core/modes.h
@@ -180,12 +180,14 @@ enum class CharmHadron : o2::analysis::femto::datatypes::CharmHadronType {
 
 enum class QvecDetector : o2::analysis::femto::datatypes::QvecDetectorType {
   kFT0C = 0,
-  kFT0A = 1
+  kFT0A = 1,
+  kQvecDetectorLast = 2
 };
 
 enum class QvecHarmonic : o2::analysis::femto::datatypes::QvecHarmonicType {
   kN2 = 2,
-  kN3 = 3
+  kN3 = 3,
+  kQvecHarmonicLast = 4
 };
 
 }; // namespace o2::analysis::femto::modes

